app/scripts/modules/core/src/pipeline/executions/execution/ExecutionMarker.tsx

```diff
@@ -81,15 +84,65 @@ export class ExecutionMarker extends React.Component<IExecutionMarkerProps, IExe
     });
   };
 
+  private manualJudgmentStatus = (stageStatus: string, manualJudgment: any) => {
+    let status = stageStatus;
+    if (manualJudgment !== undefined && manualJudgment.length && stageStatus === 'running') {
+      const existStages = this.props.stage.stages.filter((stage) => stage.status.toLowerCase() === 'running');
+      existStages.forEach(({ context, type }) => {
+        if (this.leafNodeExist(context.executionId, manualJudgment) && type === 'pipeline') {
+          status = 'waiting';
+        }
+      });
```

Comment:
You can probably use filter for all of these conditionals since you are a just seeing if any stages match these criteria.
`status === 'running' && type === 'pipeline' && this.leafNodeExist(stage.context.executionId, manualJudgement)`